apps/homescreen/js/apps.js

```diff
@@ -844,19 +985,59 @@
       this.updateSelectedIcon(null);
     },
 
+    elementName: function(element) {
+      if (!element || !(element instanceof HTMLElement)) {
+        return 'none';
+      }
+
+      var child = element.firstElementChild;
+      return child.localName === 'homescreen-group' ?
+        'group' : child.name;
+    },
+
+    isGroup: function(element) {
+      return (element && element.firstElementChild &&
+        element.firstElementChild.localName === 'homescreen-group') ?
+        true : false;
```

Comment:
This sometimes returns null or things that aren't strictly true or false, which throws off tests. I'd like to keep it explicit.